app/page.tsx

```diff
@@ -47,49 +47,69 @@ async function getASRouterLocalMessageInfoFromFile(): Promise<FxMSMessageInfo[]>
   return messages;
 }
 
-async function getMsgExpRecipeCollection(): Promise<NimbusRecipeCollection> {
-
-  const recipeCollection = new NimbusRecipeCollection()
-  await recipeCollection.fetchRecipes()
-  console.log('recipeCollection.length = ', recipeCollection.recipes.length)
-
-
-  // XXX should move to nimbusRecipe
-  function isExpRecipe(recipe : NimbusRecipe) : boolean {
-    return !recipe._rawRecipe.isRollout
-  }
-
-  const expOnlyCollection = new NimbusRecipeCollection()
-  expOnlyCollection.recipes = recipeCollection.recipes.filter(isExpRecipe)
-  console.log('expOnlyCollection.length = ', expOnlyCollection.recipes.length)
+async function getMsgExpRecipeCollection(
+  recipeCollection: NimbusRecipeCollection
+): Promise<NimbusRecipeCollection> {
+  const expOnlyCollection = new NimbusRecipeCollection();
+  expOnlyCollection.recipes = recipeCollection.recipes.filter((recipe) =>
+    recipe.isExpRecipe()
+  );
+  console.log("expOnlyCollection.length = ", expOnlyCollection.recipes.length);
 
+  const msgExpRecipeCollection = new NimbusRecipeCollection();
+  msgExpRecipeCollection.recipes = expOnlyCollection.recipes.filter((recipe) =>
+    recipe.usesMessagingFeatures()
+  );
+  console.log(
+    "msgExpRecipeCollection.length = ",
+    msgExpRecipeCollection.recipes.length
+  );
 
-  // XXX should move to nimbusRecipe
-  function isMsgRecipe(recipe : NimbusRecipe) : boolean {
-    return recipe.usesMessagingFeatures()
-  }
+  return msgExpRecipeCollection;
+}
 
-  const msgExpRecipeCollection = new NimbusRecipeCollection()
-  msgExpRecipeCollection.recipes =
-    expOnlyCollection.recipes.filter(isMsgRecipe)
-  console.log('msgExpRecipeCollection.length = ', msgExpRecipeCollection.recipes.length)
+async function getMsgRolloutCollection(
+  recipeCollection: NimbusRecipeCollection
+): Promise<NimbusRecipeCollection> {
+  const msgRolloutRecipeCollection = new NimbusRecipeCollection();
+  msgRolloutRecipeCollection.recipes = recipeCollection.recipes.filter(
+    (recipe) => recipe.usesMessagingFeatures() && !recipe.isExpRecipe()
+  );
+  console.log(
+    "msgRolloutRecipeCollection.length = ",
+    msgRolloutRecipeCollection.recipes.length
+  );
 
-  return msgExpRecipeCollection
+  return msgRolloutRecipeCollection;
 }
 
 export default async function Dashboard() {
+  const recipeCollection = new NimbusRecipeCollection()
+  await recipeCollection.fetchRecipes()
+  console.log('recipeCollection.length = ', recipeCollection.recipes.length)
+
   // XXX await Promise.all for both loads concurrently
```

Comment:
If you could change this comment to refer to Promise.allSettled, and file a bug to implement this speedup, that would be very helpful.